### PR TITLE
Default login

### PR DESCRIPTION
**Pull request by allenaerostar**
Login and logout with test cases.

### CODE REVIEW

File: dietstory-routing/dietstory-routing/urls.py [hunk 1]
```diff
@@ -18,5 +18,5 @@
 
 urlpatterns = [
     path('accounts/', include('registration.urls')),
-    path('admin/', admin.site.urls),
+    #path('admin/', admin.site.urls),
```
Comment (BenjixD) [inline dietstory-routing/dietstory-routing/urls.py]: remove this

File: dietstory-routing/registration/admin.py [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+from .models import Accounts, AccountsManager
+
+#class CustomUserAdmin(UserAdmin):
+ #   model = Accounts
 
 # Register your models here.
-admin.site.register(Accounts)
+#admin.site.register(Accounts)
+#admin.site.register(Accounts, AccountsManager)
+
```
Comment (BenjixD) [inline dietstory-routing/registration/admin.py]: remove file

File: dietstory-routing/registration/models.py [hunk 3]
```diff
@@ -29,17 +32,44 @@ def get_db_prep_value(self, value, connection, prepared=False):
...
+    def _create_user(self, username, password, email, birthday, admin_level):
+        if not username:
+            raise ValueError("Username must be set.")
+        if not email:
+            raise ValueError("Email must be set.")
+        if not password:
+            raise ValueError("Password must be set")
+        if not birthday:
+            raise ValueError("Birthday must be set.")
+
```
Comment (BenjixD) [inline dietstory-routing/registration/models.py]: private functions use double underscore __ if you don't want python `dir` to pick the method up

File: dietstory-routing/registration/models.py [hunk 4]
```diff
@@ -29,17 +32,44 @@ def get_db_prep_value(self, value, connection, prepared=False):
...
+class Accounts(AbstractBaseUser, PermissionsMixin):
     name = models.CharField(unique=True, max_length=13)
     password = models.CharField(max_length=128)
     salt = models.CharField(max_length=128, blank=True, null=True)
     pin = models.CharField(max_length=10, blank=True, null=True)
     pic = models.CharField(max_length=26, blank=True, null=True)
     loggedin = models.IntegerField(default=0, validators=[MinValueValidator(0), MaxValueValidator(250)])
-    lastlogin = models.DateTimeField(blank=True, null=True)
+    last_login = models.DateTimeField(db_column='lastlogin', blank=True, null=True)
```
Comment (BenjixD) [inline dietstory-routing/registration/models.py]: how come we need to specify the db column here?

File: dietstory-routing/registration/models.py [hunk 5]
```diff
@@ -64,7 +94,28 @@ class Accounts(models.Model):
     hwid = models.CharField(default='', max_length=12)
     fly = models.IntegerField(default=0, validators=[MinValueValidator(0), MaxValueValidator(250)])
     verified = models.IntegerField(default=0, validators=[MinValueValidator(0), MaxValueValidator(250)])
+    adminlevel = models.IntegerField(db_column='adminlevel', default=0, validators=[MinValueValidator(0)])
```
Comment (BenjixD) [inline dietstory-routing/registration/models.py]: make it consistent with other lines

File: dietstory-routing/registration/models.py (hunk 3 above)
Comment (allenaerostar) [reply dietstory-routing/registration/models.py]: Good point!

**Pull request merged**